### PR TITLE
added keyed set to stringset

### PR DESCRIPTION
**Pull request by ljsinclair**
added datatype mapping from sql-datatypes
deleted sql-datatypes
deleted cloud-data-modeling data-types

### CODE REVIEW

File: sql-preview/data-types/_posts/2022-10-25-data-type-stringset.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ STRINGSET [TIMEQUANTUM {value} [TTL '{value}}']]
 
 ## Additional information
 
+* STRINGSET has a keyed set internal datatype
```
Comment (gthrone25) [inline sql-preview/data-types/_posts/2022-10-25-data-type-stringset.md:22]: > * STRINGSET has a `keyed set` internal datatype

**Review by gthrone25 — CHANGES_REQUESTED** [review]
@ljsinclair we need to give string, id, and idset pages the same treatment since you added this for the stringset page

File: sql-preview/data-types/_posts/2022-10-25-data-type-stringset.md (hunk 1 above)
Comment (ljsinclair) [reply sql-preview/data-types/_posts/2022-10-25-data-type-stringset.md:22]: Sorted and committing :D

**Review by gthrone25 — COMMENTED** [review]
All looks good except 3 datatypes pages are still missing their updates like stringset got:
| string | keyed mutex |
| id | mutex |
| idset | set |

**Comment by ljsinclair** [comment]
Fixed the internal data types and tidied the explanations

File: sql-preview/data-types/_posts/2022-11-10-data-type-idset.md [hunk 2]
```diff
@@ -19,11 +19,14 @@ IDSET [TIMEQUANTUM {value} [TTL '{value}}']]
 
 ## Additional information
 
-* IDSET data has one standard view by default unless a timeQuantum is set
-* IDSET is used for:
+The IDSET data type:
+* has a `mutex` internal data type
```
Comment (gthrone25) [inline sql-preview/data-types/_posts/2022-11-10-data-type-idset.md:23]: needs to be `set`

File: sql-preview/data-types/_posts/2022-10-30-data-type-string.md [hunk 3]
```diff
@@ -16,9 +16,11 @@ STRING
 
 ## Additional information
 
-STRING data types work best when:
-* Looking for discrete values,
-* `group by` where cardinality is low
+The STRING data type:
+* has a `keyed-mutex` internal data type
```
Comment (gthrone25) [inline sql-preview/data-types/_posts/2022-10-30-data-type-string.md:20]: no `-` to be consistent with table

**Review by gthrone25 — COMMENTED** [review]
@ljsinclair two typos

File: sql-preview/data-types/_posts/2022-11-10-data-type-idset.md (hunk 2 above)
Comment (ljsinclair) [reply sql-preview/data-types/_posts/2022-11-10-data-type-idset.md:23]: done

File: sql-preview/data-types/_posts/2022-10-30-data-type-string.md (hunk 3 above)
Comment (ljsinclair) [reply sql-preview/data-types/_posts/2022-10-30-data-type-string.md]: done

**Review by gthrone25 — APPROVED** [review]
LGTM

**Pull request merged**